apps/explorer/lib/explorer/chain/import/runner/celo_validator_groups.ex

```diff
@@ -0,0 +1,112 @@
+defmodule Explorer.Chain.Import.Runner.CeloValidatorGroups do
+    @moduledoc """
+    Bulk imports Celo validator groups to the DB table.
+    """
+  
+    require Ecto.Query
+  
+    alias Ecto.{Changeset, Multi, Repo}
+    alias Explorer.Chain.{Import, CeloValidatorGroup}
+  
+    import Ecto.Query, only: [from: 2]
+  
+    @behaviour Import.Runner
+  
+    # milliseconds
+    @timeout 60_000
+  
+    @type imported :: [CeloValidatorGroup.t()]
+  
+    @impl Import.Runner
+    def ecto_schema_module, do: CeloValidatorGroup
+  
+    @impl Import.Runner
+    def option_key, do: :celo_validator_groups
+  
+    @impl Import.Runner
+    def imported_table_row do
+      %{
+        value_type: "[#{ecto_schema_module()}.t()]",
+        value_description: "List of `t:#{ecto_schema_module()}.t/0`s"
+      }
+    end
+  
+    @impl Import.Runner
+    def run(multi, changes_list, %{timestamps: timestamps} = options) do
+      insert_options =
+        options
+        |> Map.get(option_key(), %{})
+        |> Map.take(~w(on_conflict timeout)a)
+        |> Map.put_new(:timeout, @timeout)
+        |> Map.put(:timestamps, timestamps)
+  
+      # Enforce ShareLocks tables order (see docs: sharelocks.md)
+      multi
+      |> Multi.run(:acquire_all_items, fn repo, _ ->
+        acquire_all_items(repo)
+      end)
+      |> Multi.run(:insert_items, fn repo, _ ->
+        insert(repo, changes_list, insert_options)
+      end)
+    end
+  
+    @impl Import.Runner
+    def timeout, do: @timeout
+  
+    defp acquire_all_items(repo) do
+      query =
+        from(
+          account in CeloValidatorGroup,
+          # Enforce ShareLocks order (see docs: sharelocks.md)
```

Comment:
Comment

Reply:
If the share locks are not relevant, shouldn't we remove the query?